File: api/src/main/java/com/mystore/manager/api/admin/payload/UserPayload.java
package com.mystore.manager.api.admin.payload;


import com.mystore.manager.api.common.payload.GlobalDataPayload;
import com.mystore.manager.api.common.payload.GlobalDatePayload;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class UserPayload extends GlobalDatePayload {
    private String username;
    private String password;
    private String firstName;
    private String lastName;
    private String phoneNumber;
    private String email;
    private MultipartFile pictureIn;
    private byte[] picture;
    private boolean isActive;
    private boolean superAdmin;
    private List<ProfilPayload> profils;
    
    // Store info
    private Integer storeId;
    private String storeCode;
    private String storeName;

    public Set<String> getPrivilegesCodes() {
        if(profils == null || profils.isEmpty()) {
            return Set.of(); // Return an empty set if profils is null or empty
        }

        return this.getProfils().stream()
                .filter(profil -> profil.getPrivileges() != null) // Ensure privileges are not null
                .flatMap(profil -> profil.getPrivileges().stream()) // Flatten privileges
                .map(GlobalDataPayload::getCode) // Map to Codes
                .collect(Collectors.toSet()); // Collect as a Set
    }
}

File: ui-admin/src/app/backend/payloads/admin/userpayload.ts
import { AdminAddressPayload } from "./adminaddresspayload";
import { ProfilPayload } from "./profilpayload";

export class UserPayload {
  id: number;

  username: string;
  password: string;

  firstName: string;
  lastName: string;
  sexe: string;
  phoneNumber: string;
  email: string;
  address: AdminAddressPayload;
  picture: any;
  disableDisplay: boolean = false;
  active: boolean = true;

  profils: Array<ProfilPayload>;
  
  // POS info
  posId: number;
  posCode: string;
  posName: string;
  
  // Super admin flag
  superAdmin: boolean = false;

  constructor() {
    this.id = null;
    
    this.username = null;
    this.password = null;
    
    this.firstName = null;
    this.lastName = null;
    this.sexe = null;
    this.phoneNumber = null;
    this.email = null;
    this.address = new AdminAddressPayload();
    this.picture = null;
    this.disableDisplay = false;
    this.profils = null;
    
    this.posId = null;
    this.posCode = null;
    this.posName = null;
    this.superAdmin = false;
  }
}
